packages/react-scripts/scripts/coalesceLocales.js

```diff
@@ -16,77 +16,107 @@ exports.coalesceLocales = paths => {
     rimraf.sync(builtLocalesDir);
   }
   fs.mkdirSync(builtLocalesDir);
-  
-  const index = fs.existsSync(`${paths.appSrc}/index.tsx`) ? `${paths.appSrc}/index.tsx` : `${paths.appSrc}/index.js`
-  
+
+  const index = fs.existsSync(`${paths.appSrc}/index.tsx`)
+    ? `${paths.appSrc}/index.tsx`
+    : `${paths.appSrc}/index.js`;
+
   const list = dependencyTree.toList({
     filename: index,
     directory: paths.appPath,
-    tsConfig: index.includes('.tsx') && `${paths.appPath}/tsconfig.json`,
+    tsConfig:
+      (index.includes('.tsx') || fs.existsSync(paths.appTsConfig)) &&
+      paths.appTsConfig,
     noTypeDefinitions: true, // optional
-    detective: {
-      es6: {
-        mixedImports: true
-      }
-    },
-    nodeModulesConfig: {
-      entry: 'module',
-    },
+    detective: { es6: { mixedImports: true } },
+    nodeModulesConfig: { entry: 'module' },
     filter: p => p.includes(paths.appSrc) || p.includes('@fs'),
   });
   const realList = list.filter(p => p.includes('locales/index.js'));
   const allLocales = {};
   const collisionReport = {
     collisions: [],
     potentialCollisions: [],
-  }
+  };
   const statsReport = {
     namespaces: new Set(),
     locales: new Set(),
     keys: new Set(),
-  }
+  };
   realList.forEach(p => {
     const dir = path.dirname(p);
     // console.log('dir', dir)
-    const locales = fs.readdirSync(dir).filter(d => !d.includes('.') && d !== 'dist');
+    const locales = fs
+      .readdirSync(dir)
+      .filter(d => !d.includes('.') && d !== 'dist');
     locales.forEach(locale => {
-      statsReport.locales.add(locale)
+      statsReport.locales.add(locale);
       const namespaces = fs
         .readdirSync(path.join(dir, locale))
         .map(d => d.split('.')[0]);
       allLocales[locale] = allLocales[locale] || {};
       namespaces.forEach(ns => {
-        statsReport.namespaces.add(ns)
+        statsReport.namespaces.add(ns);
         const strings = jsonfile.readFileSync(
           `${path.join(dir, locale, ns)}.json`
         );
         allLocales[locale][ns] = allLocales[locale][ns] || {};
         // console.log('strings', Object.keys(strings))
         // console.log('allLocales', Object.keys(allLocales[locale][ns]||{}))
-        Object.keys(strings).map(key => statsReport.keys.add(key))
-        const collisions = _.intersection(Object.keys(allLocales[locale][ns]), Object.keys(strings))
-        if (collisions && collisions.length>0) {
+        Object.keys(strings).map(key => statsReport.keys.add(key));
+        const collisions = _.intersection(
+          Object.keys(allLocales[locale][ns]),
+          Object.keys(strings)
+        );
+        if (collisions && collisions.length > 0) {
```

Comment:
intersection always returns an array, not sure why we need this check now?

Reply:
I didn't change the logic here.